training/architecture/12_FACTOR_APP_PRINCIPLES.md

```diff
@@ -0,0 +1,73 @@
+12 Factor App Principles
+===========================
+
+In Digital Publishing we build, deploy and manage applications which run as services. The twelve-factor methodology helps us follow best practices and build applications which are portable and resilient. This module will introduce you to the Twelve-factor principles and provide you with examples of how we use them in our team. 
+
+Note: The examples refer primarily to Go apps because that's what I know about. Please add more details or alternative examples which might be useful.
+
+## Pre reading
+
+- [About version control](https://git-scm.com/book/en/v2/Getting-Started-About-Version-Control)
+- [What is Continuous Integration (CI) and Continuous Delivery (CD)?](https://www.redhat.com/en/topics/devops/what-is-ci-cd)
+
+
+## Pre requisites
+None.
+
+## Materials
+Familiarise yourself [The Twelve-Factor App Principles](https://12factor.net/).
+
+In this section you can find out how we apply each principle in Digital Publishing:
+
+### 1. Codebase
+We create one GitHub repository per application, for example the [dp-bulletin-api](https://github.com/ONSdigital/dp-bulletin-api) repository contains our codebase for that app. 
```

Comment:
Worth adding a description of what makes an app a 12 factor app - A twelve-factor app is always tracked in a version control system

Reply:
Thank you for all the suggestions, they helped me understand the principles better (I think!)